From 46630132062b27ae6ff66ba0856fcc3c5b8f8bab Mon Sep 17 00:00:00 2001
From: Nikhil Badola <nikhil.badola@freescale.com>
Date: Sun, 19 Oct 2014 18:55:07 +0530
Subject: [PATCH 370/741] usb: fsl: Set USB_EN bit to select ULPI phy

commit ec4f62a71f9c34b26933d6972bf99b4f8db68fcb from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Set USB_EN bit to select ULPI phy for USB controller version 2.5

Signed-off-by: Nikhil Badola <nikhil.badola@freescale.com>
Signed-off-by: yinbo.zhu <yinbo.zhu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/usb/host/ehci-fsl.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/usb/host/ehci-fsl.c b/drivers/usb/host/ehci-fsl.c
index 55f86b716bc5..7eb03848e2aa 100644
--- a/drivers/usb/host/ehci-fsl.c
+++ b/drivers/usb/host/ehci-fsl.c
@@ -196,6 +196,12 @@ static int fsl_ehci_drv_probe(struct platform_device *pdev)
 		tmp |= 0x4;
 		iowrite32be(tmp, hcd->regs + FSL_SOC_USB_CTRL);
 	}
+
+        /* Set USB_EN bit to select ULPI phy for USB controller version 2.5 */
+        if (pdata->controller_ver == FSL_USB_VER_2_5 &&
+                pdata->phy_mode == FSL_USB2_PHY_ULPI)
+                iowrite32be(USB_CTRL_USB_EN, hcd->regs + FSL_SOC_USB_CTRL);
+	
 	/*
 	 * Enable UTMI phy and program PTS field in UTMI mode before asserting
 	 * controller reset for USB Controller version 2.5
-- 
2.17.1

